cub/cub/thread/thread_reduce.cuh

```diff
@@ -102,23 +169,48 @@ ThreadReduce(T* input, ReductionOp reduction_op, PrefixT prefix, Int2Type<LENGTH
  *
  * @param[in] reduction_op
  *   Binary reduction operator
- *
- * @param[in] prefix
- *   Prefix to seed reduction with
  */
+template <int LENGTH, typename T, typename ReductionOp>
+_CCCL_DEVICE
+_CCCL_FORCEINLINE ::cuda::std::__enable_if_t<(!enable_dpx_reduction<LENGTH, T, ReductionOp>() && LENGTH > 1), T>
+ThreadReduce(T* input, ReductionOp reduction_op)
+{
+  T prefix = input[0];
+  return ThreadReduce(input + 1, reduction_op, prefix, Int2Type<LENGTH - 1>{});
+}
+
+/// Specialization for DPX reduction
+template <int LENGTH, typename T, typename ReductionOp>
+_CCCL_NODISCARD _CCCL_DEVICE
+_CCCL_FORCEINLINE ::cuda::std::__enable_if_t<enable_dpx_reduction<LENGTH, T, ReductionOp>(), T>
+ThreadReduce(T* input, ReductionOp reduction_op)
+{
+  constexpr auto IS_MIN = ::cuda::std::is_same<ReductionOp, cub::Min>::value;
+  using DpxReduceOp     = ::cuda::std::_If<IS_MIN, DpxMin<T>, DpxMax<T>>;
+  auto unsigned_input   = reinterpret_cast<unsigned*>(input);
+  auto simd_reduction   = ThreadReduce<LENGTH / 2>(unsigned_input, DpxReduceOp{});
+  T simd_values[2]; // TODO (fbusato): use bit_cast
```

Comment:
Suggestion: I assume there is a reason you haven't used `bit_cast` here? Are you just waiting for #2258 or is there another reason? Consider adding that reason to the comment, so when somebody walks by that place in the future, they can consider updating it.

Reply:
#2258 got merged now.